firewall/handlers/handlers_util.go

```diff
@@ -1,6 +1,7 @@
 package handlers
 
 func removeDuplicates(xs *[]string) {
```

Comment:
what about to return result as new slice with unique values? anyway here memory consumption not decreased and in both cases where we use `removeDuplicates` we copy all values and trash at end of slice `*xs = (*xs)[:j]` it doesn't create new slice, it just move `len` fieldn in slice structure, capacity of slice left unchanged

Reply:
It doesn't matter for me actually. I have found this code fragment on stackoverflow and it had >20 upvotes so I decided not to create a wheel. If you really think it needs to be refactored, I will do it

Reply:
will be good and code will look simpler